okhttp/src/main/java/okhttp3/internal/http/HttpEngine.java

```diff
@@ -637,17 +581,14 @@ public void readResponse() throws IOException {
 
       httpStream.writeRequestHeaders(request);
 
-      //Update the networkRequest with the possibly updated interceptor request.
-      networkRequest = request;
```

Comment:
Basically we used to snapshot in the network request at the time of the network I/O. Now we just accept whatever network request the network interceptors return.